modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastRefresher.kt

```diff
@@ -0,0 +1,135 @@
+package au.com.shiftyjelly.pocketcasts.repositories.sync
+
+import android.os.SystemClock
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.entity.Podcast
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.models.type.EpisodePlayingStatus
+import au.com.shiftyjelly.pocketcasts.models.type.EpisodeStatusEnum
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.EpisodeManager
+import au.com.shiftyjelly.pocketcasts.servers.extensions.notModified
+import au.com.shiftyjelly.pocketcasts.servers.extensions.wasCached
+import au.com.shiftyjelly.pocketcasts.servers.podcast.PodcastCacheServiceManager
+import au.com.shiftyjelly.pocketcasts.utils.DateUtil
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import com.automattic.android.tracks.crashlogging.CrashLogging
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import javax.inject.Inject
+
+class PodcastRefresher @Inject constructor(
+    private val episodeManager: EpisodeManager,
+    private val appDatabase: AppDatabase,
+    private val cacheServiceManager: PodcastCacheServiceManager,
+    private val crashLogging: CrashLogging,
+) {
+    suspend fun refreshPodcast(existingPodcast: Podcast, playbackManager: PlaybackManager) {
+        try {
+            val podcastResponse = cacheServiceManager.getPodcastResponse(existingPodcast.uuid)
+            if (podcastResponse.wasCached() || podcastResponse.notModified()) {
```

Comment:
With my tests, the `wasCached` method never thought this call was cached. If the app already has the correct version, the server will return a 304 status code. I have added this as an extra check.